### PR TITLE
Default health check to master in development

### LINKED ISSUE #2854: Latest commit remains undefined in local development
On the dev build of Telescope, the commit link still shows undefined, which is strange considering that we fixed this awhile ago. The reasoning for this lies [here](https://github.com/Seneca-CDOT/telescope/blob/master/src/backend/web/routes/health.js#L7).

When building Telescope locally, the frontend uses the legacy backend's url as the API_URL. It looks for a health check route that exists. This route tried to obtains the sha variable via an env value that doesn't exist anymore. This is not an issue on staging or production since the sha is obtained differently than how it is here. 

Since I'm pretty sure we don't plan on adding dev checks to the status service, this line will need to be fixed up.

### PR DESCRIPTION
**Pull request by HyperTHD**
## Issue This PR Addresses
This PR closes #2854 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

When looking at Telescope in development build, the git `sha` remained undefined, even though we fixed it for staging and production. The health route in the backend was the root cause, since it's the backend that's used as the `API` route in the local telescope build. This PR defaults the sha to point to the current commit on the master branch.

## Steps to test the PR

The link that points to the current `sha` should point to the current commit on master. It shouldn't have any specific sha hash, but should refer to master only.

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/backend/web/routes/health.js
Comment (manekenpix) [inline src/backend/web/routes/health.js:7]:
```suggestion
const sha = process.env.GIT_COMMIT || 'master';
```

**Comment by manekenpix** [comment]
I don't know if you'd like to solve the whole issue here or in a follow-up PR, but the thing is that the `backend` folder will probably be gone some time in the future, so we should move the health check to the `status` microservice (for example). The change would imply duplicating the `health` endpoint in `status` to serve the `sha` and version, and also making the front-end fetch the `sha` and version from `status`.

File: src/backend/web/routes/health.js
Comment (HyperTHD) [reply src/backend/web/routes/health.js:7]: This env value doesn't exist anymore in any of our env files, should we really be referring to this still?

**Comment by HyperTHD** [comment]
I don't mind solving it here, since it makes alot of sense to grab it from the status service instead of the backend.

**Comment by HyperTHD** [comment]
I made a simple health check route that does what the health route should do. There's some duplicate info that I'm confused with in terms of how it's appearing. Also, ik that the route needs to be asynchronous which is not really what this new route should do. Looking for some feedback as always!

File: src/backend/web/routes/health.js
Comment (manekenpix) [reply src/backend/web/routes/health.js:7]: We add it to the `env` file used for deployment at build time [here](https://github.com/Seneca-CDOT/telescope/blob/da0b0f664743f54a22e0f2605aba82aed11d0425/tools/autodeployment/deploy.sh#L34)

Comment (HyperTHD) [reply src/backend/web/routes/health.js:7]: Ahh, figured it gets used in deployment. Will add it back

File: src/api/status/src/server.js
Comment (manekenpix) [inline src/api/status/src/server.js:11]: I don't think this is going to work. The legacy back-end has access to this file, but microservices are not really aware of what's outside of their container, so the main `package.json` won't be available.
We need to find a different way to get Telescope's version. Also, I think the front-end would need some changes. If you think it's too much for this PR, you can do it in a follow-up.

Comment (HyperTHD) [reply src/api/status/src/server.js:11]: Can definitely do the frontend portion in a follow up, also forgot the purpose of microservices, I'll figure out how to grab the version another way

**Comment by Kevan-Y** [comment]
@HyperTHD , the code looks good. But I don't think there is a way to test it until we land it right?

File: src/api/status/package.json
Comment (humphd) [inline src/api/status/package.json:34]: `2.0.0`, no `^`

File: src/api/status/src/server.js
Comment (humphd) [inline src/api/status/src/server.js:55]: Can we cache this result?  Will it call it every time we hit the health check?  Seems like we should do it once only.

**Review by humphd — DISMISSED** [review]

**Review by Kevan-Y — DISMISSED** [review]
Need to squash those commits then good to go. Thanks @HyperTHD

**Review by humphd — DISMISSED** [review]

**Review by TueeNguyen — APPROVED** [review]

**Comment by TueeNguyen** [comment]
@HyperTHD last rebase!

**Comment by HyperTHD** [comment]
Will merge in once rebase and squash is finished. Also: 

> I don't think this is going to work. The legacy back-end has access to this file, but microservices are not really aware of what's outside of their container, so the main `package.json` won't be available. We need to find a different way to get Telescope's version. Also, I think the front-end would need some changes. If you think it's too much for this PR, you can do it in a follow-up.

I'll look into this further and see what changes, if any, will be needed and will do them in a follow-up PR

**Pull request merged**
